### PR TITLE
Documentation for compatibility with underscored versions

### PR DESCRIPTION
**Pull request by lionel-**

### CODE REVIEW

File: vignettes/compatibility.Rmd [hunk 1]
```diff
@@ -79,7 +79,161 @@ Simply copy the results of this function in your package.
...
+lazyeval::interp(~ mean(var), var = rlang::sym("mpg"))
+```
+
+You would now unquote:
+
+```{r, eval=FALSE}
+var <- "mpg"
+quo(mean(!! rlang::sym(var)))
+```
+
```
Comment (hadley) [inline vignettes/compatibility.Rmd]: Add pointer to programming vignette here

File: vignettes/compatibility.Rmd [hunk 2]
```diff
@@ -79,7 +79,161 @@ Simply copy the results of this function in your package.
...
+have classes `c("sf", "data.frame")`. Authors of such packages should
+provide a method for the non-underscored generic in order to be
+compatible with dplyr:
+
+```{r, eval = FALSE}
+filter.sf <- function(.data, ...) {
+  st_as_sf(NextMethod())
+}
+```
+
```
Comment (hadley) [inline vignettes/compatibility.Rmd]: If you need help with this, please let us know!

Comment (lionel-) [inline vignettes/compatibility.Rmd]: @hadley should this comment be added to the vignette?

Comment (hadley) [inline vignettes/compatibility.Rmd]: Yes please

**Pull request merged**